### PR TITLE
Add support for TiDB

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

File: orm/db_tidb.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// Copyright 2014 beego Author. All Rights Reserved.
```
Comment (wallclockbuilder) [inline orm/db_tidb.go]: Increment copyright year to 2015.
Code is being modified in 2015

**Comment by astaxie** [comment]
@ngaut please send the PR to develop branch. Thanks

File: orm/db_tidb.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	return "SELECT table_name FROM information_schema.tables WHERE table_type = 'BASE TABLE' AND table_schema = DATABASE()"
+}
+
+// show columns sql of table for TiDB.
+func (d *dbBaseTiDB) ShowColumnsQuery(table string) string {
+	return fmt.Sprintf("SELECT COLUMN_NAME, COLUMN_TYPE, IS_NULLABLE FROM information_schema.columns "+
+		"WHERE table_schema = DATABASE() AND table_name = '%s'", table)
+}
+
+// execute sql to check index exist.
```
Comment (wallclockbuilder) [inline orm/db_tidb.go]: exit -> exists

File: orm/orm_test.go [hunk 3]
```diff
@@ -986,6 +990,11 @@ func TestValuesFlat(t *testing.T) {
 }
 
 func TestRelatedSel(t *testing.T) {
+	if IsTidb {
+		// Tidb do not support foreign key now
```
Comment (wallclockbuilder) [inline orm/orm_test.go]: 'does not'

File: orm/qb_tidb.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
+// Copyright 2014 beego Author. All Rights Reserved.
```
Comment (wallclockbuilder) [inline orm/qb_tidb.go]: Increase copyright year to 2015

**Pull request closed without merging**